### PR TITLE
Added helper method for calculating shellbar height

### LINKED ISSUE #3266: Splitview not opening with topnav header disabled
Set **settings.header.disabled** to true, and try opening the splitview

`Luigi.navigation().openAsSplitView('/home')`

Following error shows.

```
bundle.js:2 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'clientHeight')
    at Object.getContentAreaHeight (bundle.js:2:269782)
    at Object.open (bundle.js:2:257734)
    at pe (bundle.js:2:614100)
    at bundle.js:2:628333
```

Error doesn't show when header disabled is false.

### PR DESCRIPTION
**Pull request by somabagyinszky7**
**Description**
Added helper method `getShellbarHeight()` for calculating the height of the `shellbar`; It also includes logic for when the `shellbar` is not present, e.g. when one sets **settings.header.disabled** to `true`.
Changes proposed in this pull request:
- Add a helper method under `GenericHelpersClass` called `getShellbarHeight()` which returns the `clientHeight` property of the HTML element, with the class `luigi-shellbar-wrapper`, if present; 0 otherwise.
- Change the `getContentAreaHeight` method in the `GenericHelperClass` to use this new helper method for getting the height of the `shellbar`.
- Change the `calculateAndSetSplitViewValues` method in the `SplitViewSvcClass` to use this new helper method for getting the height of the `shellbar`
- change the `updateSizes` method in `SplitView.svelte` to use this new helper method for getting the height of the `shellbar`

**Related issue(s)**
Fixes #3266

### CODE REVIEW

**Comment by ndricimrr** [comment]
Hi @somabagyinszky7 ,
Thanks for the contribution. Seems like the unit tests are failing. Could you have a look at that. 
You can run the tests as follows:
```
cd core
npm run test
```

**Comment by somabagyinszky7** [comment]
Hi @ndricimrr,
Thank you for your message. Indeed I forgot to update the unit test file to use the new helper method in the tests.
However, they are still failing here. Is it a cache issue?
The locally run tests are all passing. Please see the screenshot below.
luigi_tests

**Comment by ndricimrr** [comment]
Tests are green now 👍 . Will have another look.

File: core/src/utilities/helpers/generic-helpers.js [hunk 1]
```diff
@@ -285,8 +292,7 @@ class GenericHelpersClass {
   }
 
   /**
-   * Returns a new Object with the same object,
-   * without the keys that were given.
+   * Returns a new Object with the same object,   * without the keys that were given.
```
Comment (ndricimrr) [inline core/src/utilities/helpers/generic-helpers.js:295]:
```suggestion
   * Returns a new Object with the same object,   
   * without the keys that were given.
```
Seems like an unnecessary change/typo

File: core/src/utilities/helpers/generic-helpers.js [hunk 2]
```diff
@@ -252,7 +252,14 @@ class GenericHelpersClass {
   }
 
   getContentAreaHeight /* istanbul ignore next */() {
-    return this.getInnerHeight() - LuigiElements.getShellbar().clientHeight;
+    const contentAreaHeight = this.getInnerHeight() - this.getShellbarHeight();
+    return contentAreaHeight;
+  }
+
+  getShellbarHeight() {
```
Comment (ndricimrr) [inline core/src/utilities/helpers/generic-helpers.js:259]: Can you add some JSDoc comment to the new function you added. Check the other examples in this file, similar to the following: 

```
  /**
   * {Function description}
   *
   * @returns {return description}
   */
```

Comment (ndricimrr) [inline core/src/utilities/helpers/generic-helpers.js:259]: To avoid future bugs, we usually add unit tests for each new function that is added/implemented. 

Could you please add a unit test for the new **getShellbarHeight** function that you added. 
Unit tests for this function should fit under `core/test/utilities/helpers/generic-helpers.spec.js`. You can check the file for similar examples.

**Review by ndricimrr — CHANGES_REQUESTED** [review]
Looks good. Just a couple of comments to be resolved

**Comment by somabagyinszky7** [comment]
Hi @ndricimrr,
Thank you for your comments! I will address them this week and will keep in mind for the future.

File: core/src/utilities/helpers/generic-helpers.js (hunk 2 above)
Comment (somabagyinszky7) [reply core/src/utilities/helpers/generic-helpers.js:265]: Added some JSDoc comments for the new function, `getShellbarHeight`. See [aaed82f](https://github.com/SAP/luigi/pull/3301/commits/aaed82f835ff90dda1e439571438c2d480c25ace).

Comment (somabagyinszky7) [reply core/src/utilities/helpers/generic-helpers.js:265]: I added unit tests for the new `getShellbarHeight` function. See [5fc1aa8](https://github.com/SAP/luigi/pull/3301/commits/5fc1aa8050f5997c0d2064e58bb77ee2d18acb29).

**Comment by ndricimrr** [comment]
Looks good. Thanks for the addition. 
Seems like there are some unrelated changes on this PR made in the `docs` and `scripts/tools` folder. I assume its a  misalignment with main branch history. Could you undo/clean those changes here so they don't cause any issue on main. 
Feel free to also create a new branch with only your changes if that's easier :)

**Review by ndricimrr — APPROVED** [review]
Looks good thanks for your contribution
